src/epics/chatEpics.js

```diff
@@ -73,98 +67,34 @@ const suggestionEpic = (action$) =>
-    const urlHash = document.location.hash + pincode;
-    // 1. Get passphrase
-    const {
-      passphrase,
-      isNewPassphrase
-    } = getPassphrase(urlHash);
-
-    // 3. Get email based on passphrase
-    let email = getEmail(passphrase);
-    // 4. Decrypt to get key pair
-    miniLock.crypto.getKeyPair(passphrase, email, (keyPair) => {
-      miniLock.session.keys = keyPair;
-      miniLock.session.keyPairReady = true;
-      let mID = miniLock.crypto.getMiniLockID(keyPair.publicKey);
-      // 4. When we have keypair, login:
-
-      if (isNewPassphrase) {
-        document.location.hash = '#' + passphrase;
-      }
-
-      observer.next({ passphrase, keyPair, mID, isNewRoom: isNewPassphrase });
-      observer.complete();
-    });
-  });
-}
-
-function createDecryptMessageObservable({ message, mID, secretKey }) {
-  return Observable.create(function (observer) {
-    miniLock.crypto.decryptFile(message, mID, secretKey,
-      function (fileBlob, saveName, senderID) {
-        const reader = new FileReader();
-        reader.addEventListener("loadend", () => {
-          const authToken = reader.result;
-          observer.next(authToken);
-          observer.complete();
-        });
-
-        reader.readAsText(fileBlob);
-      });
+const initChatHandlerConnection = ({ authToken, secretKey, mID, isNewRoom }) => {
+  return chatHandler.initConnection({
+    authToken,
+    secretKey: secretKey,
+    mID
   });
-}
-
-function getAuthRequestSettings({ mID }) {
-  const settings = {
-    url: authUrl,
-    responseType: 'blob',
-    headers: {
-      'X-Minilock-Id': mID
-    },
-    method: 'GET'
-  };
-  return settings;
-}
+};
 
 const initConnectionEpic = (action$) =>
   action$.ofType(CHAT_INIT_CONNECTION)
-    .mergeMap(createKeyPairObservable)
-    .mergeMap(({ keyPair, mID, isNewRoom, passphrase }) =>
-      ajax(getAuthRequestSettings({ mID }))
-        .map(ajaxResult => ({
-          message: ajaxResult.response,
-          mID,
-          secretKey: keyPair.secretKey
-        }))
-        .mergeMap(createDecryptMessageObservable)
-        .mergeMap(authToken =>
-          chatHandler.initConnection({
-            authToken,
-            secretKey: keyPair.secretKey,
-            mID
-          })
-        )
-        .mergeMap(() =>
-          Observable.merge(
-            chatHandler.getMessageSubject()
-              .map(addMessage),
-
-            chatHandler.getUserStatusSubject()
-              .map(setUserStatus),
-
-            Observable.of(connectionInitiated(), alertSuccess(`${isNewRoom ? 'New room created.' : ''} Connected to server.`))
-          )
-        ).catch(error => {
-          console.error(error);
-          return Observable.from([
-            alertWarning('Lost connection to chat server. Trying to reconnect...'),
-            disconnected()
-          ]);
-        })
-    )
+    .mergeMap(initChatHandlerConnection)
+    .mergeMap((isNewRoom) =>
+      Observable.merge(
```

Comment:
Step 8: with `chatHandler` initialized and connected, do all the same stuff we did before.